### ISSUE TITLE
fix(model): refuse LayerNorm models on the RMSNorm-only compute HAL

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12607]
# fix(model): refuse LayerNorm models on the RMSNorm-only compute HAL

```routing
lane: model
paths: ["internal/model/qwen35_backend_contract.go", "internal/model/qwen35_backend_contract_test.go"]
expected_steps: 4
```

## Current state

At source snapshot `6872f998b465d96633c3356ef2382c97fc98cc7c`,
`NewBackendSessionChecked` describes a fail-closed architecture/backend admission gate, but
`ValidateBackendForwardConfig` returns success for every non-Qwen3.5 configuration. The generic
HAL then calls `Backend.RMSNorm` for both block norms and the final norm, even when
`Config.LayerNorm` requires mean subtraction. A valid two-layer synthetic LayerNorm model changes
one final logit by `0.15222335` between the legacy CPU reference and the admitted `cpu-ref` HAL.

Exact seams at the pinned source are:

- `internal/model/config.go:90-92` (`Config.LayerNorm`) defines mean-subtracting LayerNorm.
- `internal/model/arch.go:584-620` (`normCfg`, `layernorm`) implements the supported legacy result.
- `internal/model/hal.go:746-757`, `788-804`, and `900-916` (`stepHALMode`) unconditionally select RMSNorm.
- `internal/model/qwen35_backend_contract.go:162-190` (`ValidateBackendForwardConfig`) admits the configuration.

Open-issue title/body search on 2026-09-09 found only broad model-onboarding issue #11085; it does
not identify this HAL numerical mismatch or request a fail-closed LayerNorm admission gate.

## Problem frame

- Centrality: Core
- P1 Context: preserved - valid loaded architecture semantics reach the same checked session constructor and cannot silently change with backend selection.
- P2 Net value: advanced - prevents serving numerically incorrect logits under the advertised CPU reference class and every generic device backend.
- P3 Adaptation: N/A - this is a local correctness gate using the existing typed refusal mechanism.
- P4 Operations: advanced - model startup fails before weight upload with an actionable reason instead of producing plausible wrong tokens.

## Why this is next

`cpu-ref` is documented as bit-identical to the model package, and `NewBackendSessionChecked` is
documented as fail closed before upload. The current mismatch violates both contracts on a tiny,
deterministic CPU-only model and therefore should be blocked before further backend optimization.

## Parent context

2026-09-09 inference core audit, pinned to `6872f998b465d96633c3356ef2382c97fc98cc7c`.

## Working spine

Loaded LayerNorm config -> checked backend admission -> typed unsupported-architecture refusal ->
no HAL weight upload -> legacy CPU model path remains available.

## Core through-line

Extend the existing checked-session validation with the smallest architecture predicate that
rejects `Config.LayerNorm` while the backend contract exposes only `RMSNorm`. Add one regression
test proving the refusal happens before session construction.

## Implementation steps

1. Add a generic-HAL architecture discriminator/error path without changing the dedicated Qwen3.5 hybrid admission contract.
2. Reject `LayerNorm=true` only when a non-nil backend would enter `stepHALMode`'s generic RMSNorm path.
3. Add typed-error tests plus nil-backend and ordinary RMSNorm admission controls.
4. Run the focused checks and the existing default HAL CPU parity witness.

## Gold-plating boundary

- Do not add a LayerNorm primitive, bias uploads, or fused LayerNorm kernels in this leaf.
- Do not change the legacy model implementation or loader-derived family flags.
- Do not combine unrelated MLP, topology, RoPE, or attention capabilities into this test.
- Do not weaken or reuse the dedicated Qwen3.5 GDN forward error for this generic-HAL refusal.

## Done condition

- [ ] `ValidateBackendForwardConfig` returns a typed `UnsupportedBackendForwardError` for `LayerNorm=true` with a non-nil generic backend.
- [ ] The refusal reason names LayerNorm versus the RMSNorm-only HAL contract.
- [ ] Ordinary RMSNorm configs and the nil-backend legacy path remain admitted.
- [ ] The focused admission test and existing HAL CPU parity test pass.
- [ ] Qwen3.5 hybrid admission retains its current structural capability checks and error identity.

## Definition of done

The checked backend-session entry point cannot execute a LayerNorm model through RMSNorm-only HAL
operators, and the unchanged legacy CPU path remains usable for that model.

## Witness

The following complete test was executed through a Go overlay, leaving the public source tree
unchanged. Save it as `${TMPDIR:-/tmp}/audit_layernorm_hal_test.go`:

```go
package model

import (
	"math"
	"testing"
	"github.com/anthony-chaudhary/fak/internal/compute"
)

func TestAuditHALLayerNormParity(t *testing.T) {
	cfg := Config{
		HiddenSize: 16, NumLayers: 2, NumHeads: 4, NumKVHeads: 2,
		HeadDim: 4, IntermediateSize: 32, VocabSize: 64,
		RMSNormEps: 1e-5, RopeTheta: 10000, TieWordEmbeddings: true,
		EOSTokenID: -1, LayerNorm: true,
	}
	m := NewSynthetic(cfg)
	prompt := []int{3, 7, 11, 19, 23}
	want := m.NewSession().Prefill(prompt)
	hal, err := m.NewBackendSessionChecked(compute.Pick("cpu-ref"))
	if err != nil { t.Fatalf("valid LayerNorm config was refused: %v", err) }
	got := hal.Prefill(prompt)
	var max float32
	at := -1
	for i := range got {
		d := float32(math.Abs(float64(got[i] - want[i])))
		if d > max { max, at = d, i }
	}
	if max != 0 {
		t.Fatalf("HAL LayerNorm parity: max|delta|=%g at logit %d; got=%g want=%g", max, at, got[at], want[at])
	}
}
```

## Verifiable Witness

From the pinned checkout root, create a repository-neutral overlay and run:

```bash
repo="$(pwd)"
repro="${TMPDIR:-/tmp}/audit_layernorm_hal_test.go"
overlay="${TMPDIR:-/tmp}/audit-layernorm.overlay.json"
printf '{"Replace":{"%s/internal/model/audit_layernorm_hal_test.go":"%s"}}\n' "$repo" "$repro" > "$overlay"
go test -overlay "$overlay" ./internal/model -run '^TestAuditHALLayerNormParity$' -count=1 -v
```

Observed CPU software failure:

```text
=== RUN   TestAuditHALLayerNormParity
    audit_layernorm_hal_test.go:29: HAL LayerNorm parity: max|delta|=0.15222335 at logit 60; got=0.13680246 want=-0.015420884
--- FAIL: TestAuditHALLayerNormParity (0.00s)
FAIL
FAIL github.com/anthony-chaudhary/fak/internal/model 0.030s
FAIL
```

The diagnostic above is intentionally a before-fix numerical proof. The landing regression must
assert the concrete LayerNorm typed rejection and its axis-specific reason, not accept an arbitrary
constructor error.

## Acceptance gate

```text
go test ./internal/model -run '^(TestValidateBackendForwardConfigRejectsLayerNorm|TestHALSessionMatchesLegacyCPUReference)$' -count=1
```

## Likely files

- `internal/model/qwen35_backend_contract.go`
- `internal/model/qwen35_backend_contract_test.go`

## Lane

`model`; two-file admission-correctness leaf, four expected steps.

## Blast radius / fallback

All non-nil compute backends currently share this wrong generic HAL path. The legacy nil-backend
session remains the quarantined functional fallback. Other packages and non-LayerNorm models are
unaffected. Coordinate changes to `qwen35_backend_contract.go` with the RoPE and sliding-window
admission leaves from the same audit; land them serially if all remain open.

## Closure binding

Close only when the resolving commit cites this issue, carries `(fak model)`, and attaches the
focused green admission witness. CPU software evidence is sufficient; no GPU claim is part of this
leaf.
